### DISCUSSION TITLE
[Feature] [Web] Add new location via search when editing metadata

### DISCUSSION THREAD

**Discussion by thariq-shanavas** [immich-app/immich#5691] [Feature Request]
### The feature

Thank you for the recently introduced "Edit Metadata" feature, and the fantastic web UI!
As of now, the only way to enter a new location is by dropping a pin on a map. I would love to be able to add a location via search (perhaps with a predictive autofill/dropdown menu like Google Maps?). 

Immich, v1.90
![image](https://github.com/immich-app/immich/assets/19273195/ee0e9636-05ba-4738-a9a7-d96ef6be919c)

Google Maps UI
![image](https://github.com/immich-app/immich/assets/19273195/3b979073-28a2-4ed0-ade3-39833c4a371d)

### Platform

- [ ] Server
- [X] Web
- [ ] Mobile

**Comment by GregTroar** [comment]
+1 need this as well

**Comment by arkus7** [comment]
It would require finding an open-source alternative or free service that can provide this type of data, as the service that Google Maps uses (Google Places API) is not free. 

Maybe an option to enable this by providing a Google Maps key would be a solution. If the admin of an instance would like to have such a feature, they could enable it via the administration panel and provide their key that the server would use to support autocomplete and fetching lat & long information from the provided location name.

**Reply by bo0tzz** [reply]
Immich keeps data for the reverse geocoding feature in the postgres database. I think that has enough information available to provide this feature as well, without needing an external API.

**Comment by bo0tzz** [comment]
Added in #7139
